src/google/protobuf/generated_message_tctable_lite.cc

```diff
@@ -2816,6 +2848,438 @@ PROTOBUF_NOINLINE const char* TcParser::MpMap(PROTOBUF_TC_PARAM_DECL) {
   PROTOBUF_MUSTTAIL return ToTagDispatch(PROTOBUF_TC_PARAM_NO_DATA_PASS);
 }
 
+inline void SetHasBit(void* x, uint64_t fd, void* dummy) {
+  using FFE = TcParseTableBase::FastFieldEntry;
+  auto idx = (fd >> FFE::kHasBitShift) & FFE::kHasBitMask;
+  ABSL_DCHECK(((fd & FFE::kCardMask) != FFE::kSingular) || idx == 0) << idx;
+  x = (fd & FFE::kCardMask) == FFE::kSingular ? dummy : x;
+  static_cast<char*>(x)[idx / 8] |= 1 << (idx & 7);
+}
+
+inline void Store(uint64_t value, void* out, uint64_t fd, void* dummy) {
+    using FFE = TcParseTableBase::FastFieldEntry;
+    unsigned offset = fd >> FFE::kOffsetShift;
+    out = static_cast<char*>(out) + offset;
+    *static_cast<bool*>(out) = value;
+    auto dst = (fd & FFE::kRep32Bit) ? out : dummy;
```

Comment:
This adds branch so this code isn't exactly branch'less, right? Is it still better than switch due to at most two branches (compared to 3 in switch)?